packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/UpdateCategoriesTests.java

```diff
@@ -0,0 +1,235 @@
+        final RestCategoryModel updatedCategory = restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(createdCategory)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(OK);
+        updatedCategory.assertThat().isEqualTo(createdCategory, IGNORE_FIELD_NAME);
+        updatedCategory.assertThat().field(FIELD_NAME).isNot(createdCategory.getName());
+        updatedCategory.assertThat().field(FIELD_NAME).is(newName);
+    }
+
+    /**
+     * Update a subcategory of root's child category
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateSubcategory_asAdmin()
+    {
+        STEP("Prepare as admin a category under root category");
+        final RestCategoryModel createdCategory = prepareCategoryUnderRoot();
+
+        STEP("Prepare as admin a subcategory of root's child category");
+        final RestCategoryModel createdSubcategory = prepareCategoryUnder(createdCategory.getId());
+
+        STEP("Update as admin newly created subcategory");
+        final String newName = getRandomName("NewCategoryName");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(newName, createdSubcategory);
+        final RestCategoryModel updatedCategory = restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(createdSubcategory)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(OK);
+        updatedCategory.assertThat().isEqualTo(createdSubcategory, IGNORE_FIELD_NAME);
+        updatedCategory.assertThat().field(FIELD_NAME).isNot(createdSubcategory.getName());
+        updatedCategory.assertThat().field(FIELD_NAME).is(newName);
+    }
+
+    /**
+     * Try to update a category as a user and expect 403 (Forbidden) in response
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_asUserAndExpect403()
+    {
+        STEP("Prepare as admin a category under root category");
+        final RestCategoryModel createdCategory = prepareCategoryUnderRoot();
+
+        STEP("Try to update as user newly created category");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(getRandomName("NewCategoryName"), createdCategory);
+        restClient.authenticateUser(dataUser.createRandomTestUser())
+            .withCoreAPI()
+            .usingCategory(createdCategory)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(FORBIDDEN);
+    }
+
+    /**
+     * Try to update a non-existing category and receive 404 (Not Found)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_usingNonExistingCategoryAndExpect404()
+    {
+        STEP("Create a fake parent category");
+        final RestCategoryModel nonExistingCategory = createCategoryModelWithIdAndName(randomUUID().toString(), getRandomName("CategoryName"));
```

Comment:
Is it possible (I suppose it is but still extremely unlikely) that `randomUUID()` will return an id of a node existing in ACS instance and this way we can have a different return status (possibly 400 instead of 404)?